src/main/kotlin/com/coder/gateway/CoderGatewayConnectionProvider.kt

```diff
@@ -129,7 +152,7 @@ class CoderGatewayConnectionProvider : GatewayConnectionProvider {
      * Return an authenticated Coder CLI and the user's name, asking for the
      * token as long as it continues to result in an authentication failure.
      */
-    private fun authenticate(deploymentURL: URL, queryToken: String?, lastToken: Pair<String, TokenSource>? = null): Pair<BaseCoderRestClient, String> {
+    private fun authenticate(deploymentURL: URL, queryToken: String?, lastToken: Pair<String, TokenSource>? = null): Pair<CoderRestClient, String> {
```

Comment:
The View argument CoderWorkspacesStepSelection requires a CoderRestClient, which is what we are returning here.  Since we only return a CoderRestClient it seemed okay.  I suppose we could up-cast the base client to a CoderRestClient as we create CoderWorkspacesStepSelection?

Reply:
Ohhhh I see, yeah this is my fault then for putting the wrong client type in `CoderWorkspacesStepSelection`.  I will fix that later, this is good for now!